Convert to HTML form

ananya-ky opened the pull request:
***Update 2025-09-01:** We got some input from @fredrika11y in APA WG (thanks!); this is ready for TAG review.*

Hello TAG,

This PR implements the transition to an HTML form, as discussed in recent TAG meetings.

You can preview the form here: http://raw.githack.com/w3ctag/accessibility-screener/web-form/index.html

As discussed, we're trying to ensure that the "a11y-tracker" label gets added to the screener results, so APA WG is aware of the proposed platform changes. Thus, two bits of text were added.

This was added to the HTML form (after the intro that's currently in the Markdown document):

> The results of the screener will be filed as a new GitHub issue in your spec's repository, which you can link from your design review request. The [Accessible Platform Architectures WG](https://www.w3.org/WAI/about/groups/apawg/) will be tagged on the screener results, so they are aware of upcoming changes to the web platform, and can prepare for future horizontal review.

Because issue submission is done from the perspective of the GitHub user who's filling in the form, it may be that sometimes they cannot add the "a11y-tracker" label. So we add this note to the top of the issues that are created:

> **Please check that the `a11y-tracker` label has been added to this issue.** Adding the label makes the [Accessible Platform Architectures WG](https://www.w3.org/WAI/about/groups/apawg/) aware of your proposal; this helps them understand upcoming changes to the web platform, and can speed up future horizontal review. If you don't have permission to add labels in this repo, please leave this text here, so a repo admin will know to add the label. If the label has been added, please remove this note.

Commit:
accessibility screener form

martinthomson commented on index.html:
```diff
@@ -0,0 +1,297 @@
+        <div class="inline-text-container">https://github.com/
+          <div class="inline-text-input">
+            <label for="org">Organisation:</label>
+            <input id="org" type="text" name="org" aria-describedby="org-desc" required />
+            <div id="org-desc">
+              e.g. "w3c"
+            </div>
+          </div>/<div class="inline-text-input">
+            <label for="repo">Repository:</label>
+            <input id="repo" type="text" name="repo" aria-describedby="repo-desc" required />
+            <div id="repo-desc">
+              e.g. "process"
+            </div>
```
I found the split between org and repo here to be a little annoying to work with.  I know that you only support GitHub presently (not Forejo or GitLab or whatever), but I think that it would be better to have a URL input field rather than ask people to split their URL up.

(You could pre-fill "w3c" as the org, I suppose, but there are enough different orgs in use that I would say not.)

Then, some lightweight validation is probably necessary.  I was able to add slashes to each of these fields, which didn't work out particularly well.  A full url could be matched against `/^https:\/\/github.com\/(\[^\/]+)\/(\[^\/]+)$` (the trailing `$` being optional; you could instead accept longer URLs and just extract what you need from `$1` and `$2`; note that you would need `/^https:\/\/github.com\/(\[^\/]+)\/(\[^\/\?]+)[\/\?]?)` because GitHub adds query parameters to names occasionally now).

martinthomson commented on index.html:
```diff
@@ -0,0 +1,297 @@
+            <li>
+              a requirement for a user to perform a specific type of
+              interaction in order to enable some aspect of the API (e.g.,
+              User Activation dependent)
```
"User Activation" (title case) sounds imperious.  Link?

jyasskin replied:
https://developer.mozilla.org/en-US/docs/Web/Security/User_activation right?

martinthomson commented on script.js:
```diff
@@ -0,0 +1,106 @@
+      });
+
+      if (selectedValue) {
+        markdown += `${count}. ${questionText} \n\n    Answer: ${selectedValue} \n\n`;
```
```suggestion
        markdown += `${count}. ${questionText}\n\n    Answer: ${selectedValue}\n\n`;
```

ananya-ky committed:
Removed an extra whitespace.

Co-authored-by: Martin Thomson <mt@lowentropy.net>

jyasskin commented on index.html:
```diff
@@ -0,0 +1,297 @@
+      treatment whilst it's still in the early stages of development. These are the areas that may have the largest
+      effect on accessibility. Those questions where you answer "YES" generally need a documented strategy for managing
+      accessibility. If you don't know how to develop that strategy, you can ask us for help. Even if you are an expert
+      or have received expert help, review can still find gaps, so considering asking us for help.
```
Nit:


```suggestion
      or have received expert help, review can still find gaps, so consider asking us for help.
```

jyasskin commented on index.html:
```diff
@@ -0,0 +1,297 @@
+
+      <fieldset>
+        <legend>
+          Is any user-presented data passed through a protocol if the spec
+          is API-only?
```
This seems like 2 separate questions (and I reordered the second):
1. Is the spec API-only?
   1. If so: Is any data passed through a protocol and then presented to a user?

But ... maybe the second question applies whether or not the spec includes non-API components? So maybe ...

```suggestion
          Does the spec define a protocol that conveys any data meant to be presented to a human?
```

jyasskin commented on index.html:
```diff
@@ -0,0 +1,297 @@
+          <input id="spec-url" type="text" name="spec-url" aria-describedby="spec-url-desc" required />
+        </div>
+        <div id="spec-url-desc">
+          <p>Please ensure this dated, or for a specific commit, for example:</p>
```
Nit:

```suggestion
          <p>Please ensure this is dated, or for a specific commit, for example:</p>
```

jyasskin commented on script.js:
```diff
@@ -0,0 +1,106 @@
+
+    let questionText = legend.textContent.trim().replace(/\n/g, " ");
+    questionText = questionText.replace(/\s+/g, " ");
+    questionText = questionText.replace(/\n/g, " "); 
```
I think this is covered by the previous regex: "\n" matches \s+.

jyasskin reviewed (approved):
Thanks!

ananya-ky committed:
Spell fix

Co-authored-by: Jeffrey Yasskin <jyasskin@gmail.com>

ananya-ky committed:
Spacing plus keyword fix

Co-authored-by: Jeffrey Yasskin <jyasskin@gmail.com>

Commit:
addressing review comments: a seperate input field for spec repo url and basic validation, fixed casing and added a url for the phrase 'user activation'

matatk committed:
Apply @jyasskin's suggestion on phrasing of API/protocol question

Co-authored-by: Jeffrey Yasskin <jyasskin@gmail.com>

Commit:
cleaning up unused css

Commit:
accepting wording change as perthe review comment

Commit:
removing redundant regex as per the review comment

Commit:
issue creation bug fix, pre select N/A for all questions

Commit:
wrapped org and repo of github repo url field under em elements

Commit:
removed em elements for org and repo

ananya-ky commented:
Thanks @martinthomson and @jyasskin, we've addressed the review comments. Let us know if you are happy with the changes and if this is ready to merge.

As a side note, we've been considering merging the first two fieldsets, but we have left them separate for now. This can be revisited in a follow-up if we decide to go ahead with the change.

jyasskin committed:
Merge branch 'main' into web-form

matatk commented on index.html:
```diff
@@ -0,0 +1,287 @@
+      This checklist is a quick mechanism to assess whether or not your proposed design may need specific accessibility
+      treatment whilst it's still in the early stages of development. These are the areas that may have the largest
+      effect on accessibility. Those questions where you answer "YES" generally need a documented strategy for managing
+      accessibility. If you don't know how to develop that strategy, you can ask us for help. Even if you are an expert
+      or have received expert help, review can still find gaps, so consider asking us for help.
```
I think the intent when this was written was for 'us' to be TAG, as this is expected to be used very early in the TAG design review process. (Of course, it works as a screener for any proposal/spec, though, and it would be great to be able to use it as such.)

The issue created by the screener (in the proposer's repo) is meant to end up with the `a11y-tracker` label - so APA _should_ (signposting!) be made aware of the proposal/spec via a tracking issue in the [a11y-tracking](https://github.com/w3c/a11y-tracking) repo. This is comparable to the [i18n self-review process](https://www.w3.org/International/review-request#selfreview) which suggests that their shorter checklist results should be filed as an issue in the proposer's repo, so the proposing group can easily follow the discussion. So APA could comment on that thread (having been alerted to it via the tracking issue).

If the group wants to contact APA directly, I think they're likely to be looking at the [How to get horizontal review section of the Guidebook](https://www.w3.org/guide/documentreview/#how-to-get-horizontal-review), which I see that APA needs to update to make it a bit clearer! (We're working on improving the UX of our implementation of the HR process - linking more closely with the TAG screener is part of that.)

I think two things need to happen:

1. APA needs to provide a [template in the a11y-request repo](https://github.com/w3c/a11y-request/issues/new/choose) that covers screener results, so groups can actively ask for input if they want to. (The 'Self-review' option there relates to the longer FAST Checklist, which we're revamping - and we should clarify that.)

2. It occurs to me that if the document being reviewed is under [Explainers by Googlers](https://github.com/explainers-by-googlers) (and, Shirley, similar repos/orgs) it won't have the `a11y-tracker` label, so we need to modify the text in the generated issue to account for that, and link to said screener request issue from point 1 above.

In both cases I think we should continue to offer emailing the public APA list as an option, but having the explicit screener results request path seems necessary and helpful.

Thank you for the suggestion, and helping me to find a hole in this process!

jyasskin replied:
While your discussion identifies several good follow-ups, I don't think any of them need to block this PR from landing. I'll merge it. Follow-up PRs are, of course, welcome.
